vlm/src/main/scala/geotrellis/contrib/vlm/RasterSource.scala

```diff
@@ -41,144 +42,176 @@ import geotrellis.util.GetComponent
-      *   this region may be larger or smaller than the footprint of the data
-      * @group resample
-      */
-    def resampleToRegion(region: RasterExtent, method: ResampleMethod = NearestNeighbor): RasterSource =
-      resample(TargetRegion(region), method)
-
-    /** Reads a window for the extent.
-      * Return extent may be smaller than requested extent around raster edges.
-      * May return None if the requested extent does not overlap the raster extent.
-      * @group read
-      */
-    @throws[IndexOutOfBoundsException]("if requested bands do not exist")
-    def read(extent: Extent, bands: Seq[Int]): Option[Raster[MultibandTile]]
-
-    /** Reads a window for pixel bounds.
-      * Return extent may be smaller than requested extent around raster edges.
-      * May return None if the requested extent does not overlap the raster extent.
-      * @group read
-      */
-    @throws[IndexOutOfBoundsException]("if requested bands do not exist")
-    def read(bounds: GridBounds, bands: Seq[Int]): Option[Raster[MultibandTile]]
-
-    /**
-      * @group read
-      */
-    def read(extent: Extent): Option[Raster[MultibandTile]] =
-        read(extent, (0 until bandCount))
-
-    /**
-      * @group read
-      */
-    def read(bounds: GridBounds): Option[Raster[MultibandTile]] =
-        read(bounds, (0 until bandCount))
-
-    /**
-      * @group read
-      */
-    def read: Option[Raster[MultibandTile]] =
-        read(extent, (0 until bandCount))
-
-    /**
-      * @group read
-      */
-    def read(bands: Seq[Int]): Option[Raster[MultibandTile]] =
-        read(extent, bands)
-
-    /**
-      * @group read
-      */
-    def readExtents(extents: Traversable[Extent], bands: Seq[Int]): Iterator[Raster[MultibandTile]] =
-        extents.toIterator.flatMap(read(_, bands).toIterator)
-
-    /**
-      * @group read
-      */
-    def readExtents(extents: Traversable[Extent]): Iterator[Raster[MultibandTile]] =
-        readExtents(extents, (0 until bandCount))
-    /**
-      * @group read
-      */
-    def readBounds(bounds: Traversable[GridBounds], bands: Seq[Int]): Iterator[Raster[MultibandTile]] =
-        bounds.toIterator.flatMap(read(_, bands).toIterator)
-
-    /**
-      * @group read
-      */
-    def readBounds(bounds: Traversable[GridBounds]): Iterator[Raster[MultibandTile]] =
-        bounds.toIterator.flatMap(read(_, (0 until bandCount)).toIterator)
-
-    /**
-     * Applies the given [[LayoutDefinition]] to the source data producing a [[LayoutTileSource]].
-     * In order to fit to the given layout, the source data is resampled to match the Extent
-     * and CellSize of the layout.
-     *
-     */
-    def tileToLayout(layout: LayoutDefinition, resampleMethod: ResampleMethod = NearestNeighbor): LayoutTileSource =
-      LayoutTileSource(resampleToGrid(layout, resampleMethod), layout)
-
-    def close = { }
+  def resampleMethod: Option[ResampleMethod]
```

Comment:
Sorry, this got auto-fixed to consistent indentation and makes this diff very ... not useful